.github/workflows/main.yml

```diff
@@ -1,32 +1,108 @@
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '14.x'
+      - name: Cache maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2
+          restore-keys: ${{ runner.os }}-m2
+      - name: Build carbon-apimgt without tests, skip AspectJ, skip portal builds
+        run: mvn clean install --file pom.xml -Dmaven.test.skip=true -Dskip.aspectj=true -Dnpm.build.skip=true
+      - name: Get carbon.apimgt.version
+        run: mvn help:evaluate -Dexpression=project.version -q -DforceStdout > ../CARBON_APIMGT_VERSION_FILE
+      - name: Print carbon.apimgt.version
+        run: |
+          echo $(cat ../CARBON_APIMGT_VERSION_FILE)
+      - name: Checkout Product-APIM
+        uses: actions/checkout@v2
+        with:
+          repository: wso2/product-apim
+          ref: refs/heads/master
+          fetch-depth: '10'
+          path: product-apim
+      - name: Build product-apim with tests
+        run: mvn clean install -Dcarbon.apimgt.version=$(cat ../CARBON_APIMGT_VERSION_FILE) -fae --file product-apim/pom.xml
+        env:
+          PRODUCT_APIM_TESTS: ${{ matrix.segment }}
+      - name: Prepare upload testng report
+        if: always()
+        run: mv product-apim/modules/integration/tests-integration/tests-backend/target/surefire-reports/TEST-TestSuite.xml TEST-TestSuite_${{ matrix.id }}.xml
+      - name: Archive testng report
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: TEST-TestSuite_${{ matrix.id }}
+          path: |
+            TEST-TestSuite_${{ matrix.id }}.xml
+          if-no-files-found: warn
 
+  show-report:
+    if: always()
+    needs: build-product
+    runs-on: 'ubuntu-latest'
     steps:
-    - name: Checkout master
-      uses: actions/checkout@v1
-    - name: Set up JDK 1.8
-      uses: actions/setup-java@v1
-      with:
-        java-version: 1.8
-    - uses: actions/setup-node@v1
-      with:
-        node-version: '14.x'
-    - name: Build carbon-apimgt with Test.
-      run: mvn clean install --file pom.xml
-    - name: Get carbon.apimgt.version
-      run: mvn help:evaluate -Dexpression=project.version -q -DforceStdout > ../CARBON_APIMGT_VERSION_FILE
-    - name: Print carbon.apimgt.version
-      run: |
-        echo $(cat ../CARBON_APIMGT_VERSION_FILE)
-    - name: Checkout Product-APIM
-      uses: actions/checkout@v1
-      with: 
-        repository: wso2/product-apim
-        ref: refs/heads/master
-    - name: Build Product-Apim with Maven
-      run: mvn clean install -Dcarbon.apimgt.version=$(cat ../CARBON_APIMGT_VERSION_FILE) -fae --file ../product-apim/pom.xml
+      - uses: actions/download-artifact@v2
+        with:
+          path: reports
+      - name: Publish Test Report
+        if: ${{ always() }}
+        uses: malinthaprasan/action-surefire-report@v1
```

Comment:
Yes the original version of the plugin (https://github.com/ScaCap/action-surefire-report) has some issues as it tries to post tests results from surefire plugin to the git action itself but since the token inside github actions doesn't have necessary privileges it fails. Because of this the reporting is failing. Therefore this plugin had to be extended to disable that and just add "echo" logs with the failing test cases.
The change is added here https://github.com/ScaCap/action-surefire-report/compare/master...malinthaprasan:master

Reply:
shall we contribute it back with a config?